src/hotspot/os/linux/os_linux.cpp

```diff
@@ -929,8 +929,11 @@ bool os::create_thread(Thread* thread, ThreadType thr_type,
 
   // Add an additional page to the stack size to reduce its chances of getting large page aligned
   // so that the stack does not get backed by a transparent huge page.
-  if (stack_size >= os::Linux::default_large_page_size() && is_aligned(stack_size, os::Linux::default_large_page_size())) {
-    stack_size += os::vm_page_size();
+  size_t default_large_page_size = os::Linux::default_large_page_size();
+  if (default_large_page_size != 0) {
+    if (stack_size >= default_large_page_size && is_aligned(stack_size, default_large_page_size)) {
```

Comment:
Nit: this could have been written as a single if-clause

Reply:
Changed it. Thanks!